kn/fotos/entities.py

```diff
@@ -333,6 +347,42 @@ def get_random_foto_for(self, user):
                 return None
             r = 1
 
+    def search(self, q, user):
+        required_visibility = self.required_visibility(user)
+        filter={
+            'path': self.mongo_path_prefix,
+            'effectiveVisibility': {'$in': tuple(required_visibility)},
+        }
```

Comment:
Dit is leesbaarder

    filter = {'path': self.mongo_path_prefix,
              'effectiveVisibility': {'$in': tuple(required_visibility)}}

`filter` is trouwens een keyword.  Het is beter hier een andere naam voor te gebruiken.  `qfilter`?